packages/components/pages/f-mfa/src/components/_tests/Mfa.test.js

```diff
@@ -80,36 +106,195 @@ describe('Mfa', () => {
+            const errorCard = wrapper.find('[data-test-id="mfa-error-page"]');
+            expect(errorCard.exists()).toBe(expected);
+            expect(warnLogSpy).toHaveBeenCalled();
+        });
+    });
+
+    describe('When calling the onFormSubmit() method', () => {
+        it.each([
+            ['show', true],
+            ['NOT show', false]
+        ])('should %s error alert when hasSubmitError is %s', async (_, hasSubmitError) => {
+            // Arrange
+            wrapper = await mountSut();
+
+            // Act
+            await wrapper.setData({ hasSubmitError });
+
+            // Assert
+            const alert = wrapper.find('[data-test-id="mfa-submit-error-alert"]');
+            expect(alert.exists()).toBe(hasSubmitError);
+        });
+
+        it.each([
+            ['show', true],
+            ['NOT show', false]
+        ])('should %s validation error when showValidationError is %s', async (_, showValidationError) => {
+            // Arrange
+            wrapper = await mountSut({ shallow: false }); // Deep mount to make sure slot is rendered
+
+            // Act
+            await wrapper.setData({ showValidationError });
+
+            // Assert
+            const validationError = wrapper.find('[data-test-id="mfa-form-validation-error"]');
+            expect(validationError.exists()).toBe(showValidationError);
+        });
+
+        describe('And the form data is valid', () => {
+            it('should call the postValidateMfaToken() method with the correct params', async () => {
+                // Arrange
+                mockPostValidateMfaToken = jest.fn(() => {}); // Add default mock to the spy
+                const expectedParams = { mfa_token: 'ABC123', otp: 'test-otp' }; // eslint-disable-line camelcase
+                wrapper = await mountSut();
+                await wrapper.setData({ otp: 'test-otp' });
+
+                // Act
+                await wrapper.vm.onFormSubmit();
+
+                // Assert
+                expect(mockPostValidateMfaToken).toHaveBeenCalledWith(expectedParams);
+            });
+
+            describe('And the submitting of the form data was unsuccessful', () => {
+                it('should log bad request if the error status is 400', async () => {
+                    // Arrange
+                    const err400 = new Error('sorry but bad request');
+                    err400.response = { status: 400 };
+                    mockPostValidateMfaToken = jest.fn(() => { throw err400; }); // Add 400 error mock to the spy
+                    wrapper = await mountSut();
+                    await wrapper.setData({ otp: 'test-otp' });
+
+                    // Act
+                    await wrapper.vm.onFormSubmit();
+
+                    // Assert
+                    expect(errorLogSpy).toHaveBeenCalledWith('Bad request when submitting MFA', err400, ['account-pages', 'mfa']);
+                    expect(wrapper.vm.isSubmitting).toBe(false);
+                });
```

Comment:
I reckon you could combine some of these tests like this:
```suggestion
                it.each(
                    ['bad request', 400, 'Bad request when submitting MFA'],
                    ['throttled request', 429, 'Throttled when submitting MFA'],
                )('should log bad request if the error status is 400', async (errorMessage, status, exceptionMessage) => {
                    // Arrange
                    const error = new Error(errorMessage);
                    error.response = { status };
                    mockPostValidateMfaToken = jest.fn(() => { throw error; });
                    wrapper = await mountSut();
                    await wrapper.setData({ otp: 'test-otp' });

                    // Act
                    await wrapper.vm.onFormSubmit();

                    // Assert
                    expect(errorLogSpy).toHaveBeenCalledWith(exceptionMessage, error, ['account-pages', 'mfa']);
                    expect(wrapper.vm.isSubmitting).toBe(false);
                });
```